pkg/controller/kataconfig/openshift_reconciler.go

```diff
@@ -694,3 +715,95 @@ func (r *ReconcileKataConfigOpenShift) monitorKataConfigInstallation() (reconcil
 
 	return reconcile.Result{}, nil
 }
+
+func getConfigClientSet() (*configv1.ConfigV1Client, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		return nil, err
+	}
+
+	configV1Client, err := configv1.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return configV1Client, err
+}
+
+func getClusterVersion() (string, error) {
+	configV0Client, err := getConfigClientSet()
+	if err != nil {
+		return "", err
+	}
+
+	clusterversion, err := configV0Client.ClusterVersions().Get("version", metav1.GetOptions{})
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return "", nil
+	}
+
+	return clusterversion.Status.Desired.Version, nil
+}
+
+func getSourceImageTag(sourceImage string) (string, error) {
+	imageslice := strings.Split(sourceImage, ":")
+	if len(imageslice) <= 1 {
+		return "", fmt.Errorf("bad format of sourceImage")
+	}
+	mytag := imageslice[len(imageslice)-1]
+
+	return mytag, nil
+}
+
+func getClientSet() (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func checkCompatibility(clusterversion string, sourceImageVersion string) (bool, error) {
+	/* map OCP version to the minimum version of the sourceImage
+	 * that can be used on it. For example if a user tries
+	 * to install a sourceImage tagged with 4.5.3 on an OCP cluster
+	 * of version 4.5.4 the installation would fail because the RPM
+	 * dependencies don't match. In this case we return false to
+	 * signal that this is not supported.
+	 * TODO: don't hardcode this here, read it from a yaml file or similar
+	 */
+	minimumVersion := make(map[string]string)
```

Comment:
May be it's worth considering https://github.com/coreos/go-semver to handle version checking?

Reply:
I didn't know about it. I'll switch to that. Thanks!